histogrammar/primitives/sparselybin.py

```diff
@@ -672,41 +672,44 @@ def n_bins(self):
         """Get number of filled bins, consistent with SparselyBin and Categorize """
         return len(self.bins)
 
-    def num_bins(self, low=None, high=None):
-        """
-        Returns number of bins from low to high, including unfilled
-
-        Possible to set range with low and high params
-
-        :param low: lower edge of range, default is None
-        :param high: higher edge of range, default is None
-        :returns: number of bins in range
-        :rtype: int
+    def _bin_range(self, low=None, high=None):
+        """Utility function for calculating bins ranges and numbers for given constraints [low, high]
+        :returns: (minBin, maxBin, numBins, minBinLeftEdge, maxBinRightEdge)
         """
+        if low is not None and high is not None and low > high:
+            raise RuntimeError('low {low} greater than high {high}'.format(low=low, high=high))
         # sparse hist not filled
         if self.minBin is None or self.maxBin is None:
-            return 0
-        # trivial case
-        if low is None and high is None:
-            return (self.maxBin - self.minBin + 1)
-        # catch weird cases
-        elif low is not None and high is not None:
-            if low > high:
-                raise RuntimeError('low {low} greater than high {high}'.format(low=low, high=high))
-        # lowest edge
+            return [0, -1, 0, self.origin, self.origin+1]
```

Comment:
Return type here is list, later it's a tuple (which is preferable for unpacking)